onsp_co2/models/carbon_factor.py

```diff
@@ -78,6 +88,81 @@ def _compute_descendant_ids(self):
         for factor in self:
             factor.descendant_ids = factor.child_ids | factor.child_ids.descendant_ids 
     
+
+    def _get_count_by_model(self, model: str):
+        """
+        Calculate the total count of carbon factors for a given model.
+
+        This function reads the data from the specified model, filters it based on the carbon_in_factor_id and carbon_out_factor_id,
+        and then calculates the total count of carbon factors for each item in the data.
+
+        Args:
+            model (str, optional): The name of the model to calculate the carbon factors for.
+
+        Returns:
+            dict: A dictionary where the keys are the carbon_in_factor_id and the values are the total count of carbon factors for that id.
+        """
+        data = self.env[model].read_group(
+            self._get_carbon_domain(), ["carbon_in_factor_id", "carbon_out_factor_id"], ["carbon_in_factor_id", "carbon_out_factor_id"], lazy=False
+        )
+        total_count = defaultdict(int)
+        for item in data:
+            total_count[item["carbon_in_factor_id"][0]] += item['__count']
+        
+        return total_count
+    
+    def _get_carbon_domain(self):
+        """
+        Generate the domain for carbon factor queries.
+
+        This function creates a domain that can be used in queries to filter data based on the carbon_in_factor_id and carbon_out_factor_id.
+
+        Returns:
+            list: A list representing the domain for carbon factor queries. The list includes a logical OR operator and two tuples, each specifying a field name and a condition.
+        """
+        return ['|', ("carbon_in_factor_id", "in", self.ids), ("carbon_out_factor_id", "in", self.ids)]
+    
+    def _generate_action(self, model: str, title: str):
+        """
+        Generate an action dictionary for the specified model and title.
+
+        This function creates an action dictionary that can be used to open a new window in the Odoo UI. The new window will display the specified model's data, filtered by the carbon domain.
+
+        Args:
+            model (str): The name of the model to display in the new window.
+            title (str): The title to display in the new window.
+
+        Returns:
+            dict: An action dictionary that can be used to open a new window in the Odoo UI.
+        """
```

Comment:
Thanks for the docstrings!
Would be nice to use the python typing also like this:

```python
def _get_count_by_model(self, model: str) -> int:
```

Reply:
fix in https://github.com/MyCityCO2/mycityco2-engine/pull/13/commits/e2646e88af9a256804cde3923390095586be254c